client-side/src/graphs/graphs.jsx

```diff
@@ -0,0 +1,313 @@
+import * as React from 'react';
+import { LineChart, ChartContainer, LinePlot, BarPlot, ChartsYAxis, ChartsXAxis, Gauge, PieChart, ChartsLegend } from '@mui/x-charts';
+import Stack from '@mui/material/Stack';
+import { useQuery, gql } from '@apollo/client';
+
+const GET_VISITED_WEBSITES = gql`
+query GetVisitedWebsites {
+visitedWebsites {
+ websiteId {
+   name
+   url
+ }
+  visitsTime {
+    activityTime
+    visitDate
+  }
+}}
+`;
+
+const GET_WEBSITE = gql`
+query website{
+  websites {
+    name
+    url
+  }
+}
+`
+const GET_USERS = gql`
+ query users{
+   users {
+     name
+     password
+     email
+    profileImage
+    visitsWebsites {
+      visitsTime {
+        activityTime
+        visitDate
+      }
+      websiteId {
+        name
+        url
+      }
+    }
+   }
+ }`
+
+
+const otherProps = {
+    width: 400,
+    height: 200,
+    sx: {
+        [`.${ChartsLegend.root}`]: {
+            transform: 'translate(20px, 0)',
+        },
+    },
+};
+
+
+const VisitedWebsitesComponent = ({ startDate, endDate, user }) => {
+    const websites = useQuery(GET_WEBSITE);
+    const visitedWebsite = useQuery(GET_VISITED_WEBSITES);
+    const users = useQuery(GET_USERS);
+    const colors = ["red", "yellow", "orange", "blue", "lightgreen"]
+    let dateStart = new Date(startDate.$d);
+    let year = dateStart.getFullYear();
+    let month = ('0' + (dateStart.getMonth() + 1)).slice(-2);
```

Comment:
1. why do you add '0' in start, what will be when the month will be 10 for example? (same for the day)
2. for joining few strings and variables please use es6 normalization:
`0${dateStart.getMonth() + 1}`
3. why you need the slice(-2)?
4. you shouldn't write code inside the component, all the logic code should be inside functions